### PR TITLE
[Clang] Fix bug the way we handle duplicate vs conflicting values with loop attribute 'code_align'

### PR DESCRIPTION
**Pull request by smanna12**
https://github.com/llvm/llvm-project/pull/70762 added support for new loop attribute [[clang::code_align()]].

This patch fixes bugs for the test cases below that misses diagnostics due to discontinue to while loop during checking duplicate vs conflicting code_align attribute values in routine CheckForDuplicateLoopAttrs().

[[clang::code_align(4)]]
[[clang::code_align(4)]]
[[clang::code_align(8)]]
for(int I=0; I<128; ++I) { bar(I); }

[[clang::code_align(4)]]
[[clang::code_align(4)]]
[[clang::code_align(8)]]
[[clang::code_align(32)]]
for(int I=0; I<128; ++I) { bar(I); }

### CODE REVIEW

File: clang/lib/Sema/SemaStmtAttr.cpp [hunk 1]
```diff
@@ -405,8 +405,8 @@ static void CheckForDuplicateLoopAttrs(Sema &S, ArrayRef<const Attr *> Attrs) {
       S.Diag((*LastFoundItr)->getLocation(), diag::err_loop_attr_conflict)
           << *FirstItr;
       S.Diag((*FirstItr)->getLocation(), diag::note_previous_attribute);
+      return;
```
[the remark this answers is not in the document]
Comment (smanna12) [reply clang/lib/Sema/SemaStmtAttr.cpp:408]: Thanks @erichkeane for reviews.

Example:
```
[[clang::code_align(4)]]
[[clang::code_align(4)]]
[[clang::code_align(8)]]
for(int I=0; I<128; ++I) { bar(I); }
```
We want the loop to continue to look past the matching 4 with 4 , and then compare the first 2 with the 8
 

Now we have this:
```
[[clang::code_align(4)]] // expected-note{{previous attribute is here}}
+  [[clang::code_align(4)]] // OK
+  [[clang::code_align(8)]] // expected-error{{conflicting loop attribute 'code_align'}}
+  for(int I=0; I<128; ++I) { bar(I); }

```

[the remark this answers is not in the document]
Comment (erichkeane) [reply clang/lib/Sema/SemaStmtAttr.cpp:408]: Ah, I see, so I think this is the incorrect patch then.  I think we want the 'return' to drop OUT of the while loop, not into the 'if'.  

This as written wont catch:

```
[[clang::code_align(4)]]
 [[clang::code_align(4)]]
 [[clang::code_align(8)]] 
 [[clang::code_align(12)]] 
for(int I=0; I<128; ++I) { bar(I); }
```

Which I would expect to error 2x.

[the remark this answers is not in the document]
Comment (smanna12) [reply clang/lib/Sema/SemaStmtAttr.cpp]: >>This as written wont catch:

Yes, test case does not give 2x errors.


>>so I think this is the incorrect patch 

Yes.

I have fixed the patch with new commit https://github.com/llvm/llvm-project/pull/87372/commits/af173397479a130599a11e8d4e1ff3f4d0f3a485. Thanks for catching!

**Review by erichkeane — APPROVED** [review]

**Comment by smanna12** [comment]
Thank you @erichkeane for reviews!

**Pull request merged**